c_binding/lsm_plugin_ipc.cpp

```diff
@@ -2933,3 +2938,72 @@ void lsm_plug_target_port_search_filter(const char *search_key,
         }
     }
 }
+
+static void get_batteries(int rc, lsm_battery **bs, uint32_t count,
+                          Value &response)
+{
+    uint32_t i = 0;
```

Comment:
On 04/28/2016 02:06 AM, Gris Ge wrote:
>> @@ -2933,3 +2938,72 @@ void lsm_plug_target_port_search_filter(const char *search_key,
>>          }
>>      }
>>  }
>> +
>> +static void get_batteries(int rc, lsm_battery **bs, uint32_t count,
>> +                          Value &response)
>> +{
>> +    uint32_t i = 0;
> 
> I intend to declare and initialize variables at the beginning of each function.
> 
> Using in-loop declaration requires `-std=c99`, or you will get error on RHEL 6 which use C90 by default:
> ```
> error: ‘for’ loop initial declarations are only allowed in C99 mode
> ```

Agreed, but this is a c++ file being compiled with a c++ compiler which
does support this.

-Tony

> 
> ---
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/libstorage/libstoragemgmt/pull/16/files/b9212a274cf79babb271268fa6f9dee41c2769b3..4de1a915fdda3c89bf07968084e8d7fe7fdf89a3#r61381537
>